backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -956,4 +964,132 @@ public void deleteBranchByCommit(Session session, Long repoId, String commitHash
       }
     }
   }
+
+  @Override
+  public void addRepositoryAttributes(
+      RepositoryIdentification repositoryId, List<KeyValue> attributesList)
+      throws InvalidProtocolBufferException, ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      RepositoryEntity repositoryEntity = getRepositoryById(session, repositoryId, true);
+      repositoryEntity.setAttributeMapping(
+          RdbmsUtils.convertAttributesFromAttributeEntityList(
+              repositoryEntity, ModelDBConstants.ATTRIBUTES, attributesList));
+      repositoryEntity.setDate_updated(Calendar.getInstance().getTimeInMillis());
+      Transaction transaction = session.beginTransaction();
+      session.saveOrUpdate(repositoryEntity);
+      transaction.commit();
+      LOGGER.debug("Repository attributes added successfully");
+    } catch (Exception ex) {
+      if (ModelDBUtils.needToRetry(ex)) {
+        addRepositoryAttributes(repositoryId, attributesList);
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  @Override
+  public void updateRepositoryAttribute(RepositoryIdentification repositoryId, KeyValue attribute)
+      throws InvalidProtocolBufferException, ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      RepositoryEntity repositoryEntity = getRepositoryById(session, repositoryId, true);
+      AttributeEntity updatedAttributeObj =
+          RdbmsUtils.generateAttributeEntity(
+              repositoryEntity, ModelDBConstants.ATTRIBUTES, attribute);
+
+      List<AttributeEntity> existingAttributes = repositoryEntity.getAttributeMapping();
+      if (!existingAttributes.isEmpty()) {
+        boolean doesExist = false;
+        for (AttributeEntity existingAttribute : existingAttributes) {
+          if (existingAttribute.getKey().equals(attribute.getKey())) {
+            existingAttribute.setKey(updatedAttributeObj.getKey());
+            existingAttribute.setValue(updatedAttributeObj.getValue());
+            existingAttribute.setValue_type(updatedAttributeObj.getValue_type());
+            doesExist = true;
+            break;
+          }
+        }
+        if (!doesExist) {
+          repositoryEntity.setAttributeMapping(Collections.singletonList(updatedAttributeObj));
+        }
+      } else {
+        repositoryEntity.setAttributeMapping(Collections.singletonList(updatedAttributeObj));
+      }
+      repositoryEntity.setDate_updated(Calendar.getInstance().getTimeInMillis());
+      Transaction transaction = session.beginTransaction();
+      session.saveOrUpdate(repositoryEntity);
+      transaction.commit();
+    } catch (Exception ex) {
+      if (ModelDBUtils.needToRetry(ex)) {
+        updateRepositoryAttribute(repositoryId, attribute);
+      } else {
+        throw ex;
+      }
+    }
+  }
+
+  @Override
```

Comment:
not required

Reply:
Remove `getRepositoryAttributes` endpoint